src/jit/simdcodegenxarch.cpp

```diff
@@ -1144,12 +1144,11 @@ void CodeGen::genSIMDIntrinsicRelOp(GenTreeSIMD* simdNode)
                 // We need two additional XMM registers as scratch.
                 regMaskTP floatRsvdRegs = (simdNode->gtRsvdRegs & RBM_ALLFLOAT);
                 assert(floatRsvdRegs != RBM_NONE);
-                assert(genCountBits(floatRsvdRegs) == 2);
+                assert(genCountBits(floatRsvdRegs) == 1);
```

Comment:
Logic on line 1144 - 1151:

We don't need floatRsvdRegs local at all here

 assert(genCountBits(floatRsvdRegs) == 1);
 regNumber tmpReg1 = genRegNumFromMask(simdNode->gtRsvdRegs)